src/client/App.jsx

```diff
@@ -1,30 +1,69 @@
-import { useEffect, useState } from 'react';
-import './App.css';
-import MovieForm from './components/MovieForm';
-import UserForm from './components/UserForm';
+import { useEffect, useState } from "react";
+import "./App.css";
+import MovieForm from "./components/MovieForm";
+import UserForm from "./components/UserForm";
 
-const apiUrl = 'http://localhost:4000';
+const apiUrl = "http://localhost:4000";
 
 function App() {
   const [movies, setMovies] = useState([]);
 
   useEffect(() => {
     fetch(`${apiUrl}/movie`)
-      .then(res => res.json())
-      .then(res => setMovies(res.data));
-  }, []);
+      .then((res) => res.json())
+      .then((res) => setMovies(res.data));
+  }, [movies]);
 
   const handleRegister = async ({ username, password }) => {
-    
+    fetch(apiUrl + "/user/register", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        username,
+        password,
+      }),
+    })
+      .then((res) => res.json())
+      .then((data) => {
+      });
   };
 
   const handleLogin = async ({ username, password }) => {
-    
+    fetch(apiUrl + "/user/login", {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({
+        username,
+        password,
+      }),
+    })
+      .then((res) => res.json())
+      .then((data) => {
+        console.log(data);
+        localStorage.setItem("token", data.data);
+        localStorage.setItem("userId", data.userId);
```

Comment:
If the client needs the user ID, you could store that in the payload of the token and then the client can use the `jsonwebtoken` package to decode the token payload:

```js
jwt.decode(localStorage.getItem('token'))
```

Reply:
So in the controller I would have something like:
```js
const token = jwt.sign({ username, id }, mySecret);
```